### PR TITLE
Add engines to browsers

### LINKED ISSUE #3264: Add engine version to browser data
I think might want to add engine versions to the browser/ data.

E.g. for Opera, Edge, or other Chromium based browsers, the version isn't always clear (see how complicated it is e.g. for Opera for Android in https://github.com/mdn/browser-compat-data/pull/1712#issuecomment-416886047)

So, likely adding "engine_version" or similar to the browser json files would be useful.
https://github.com/mdn/browser-compat-data/blob/master/browsers/opera.json

Later on, these engine versions might be used to populate / validate version data for web features.

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
Fixes #3264.  This adds engine name and version fields to browsers, and adds the corresponding data to all main browsers (Chrome, Firefox, Safari, Opera, IE, and Edge, including their mobile counterparts).  The data may not be 100% accurate for older versions of Opera, Safari, and Webview Android, due to a lack of credible sources.

This does not add engine versions for other Chromium-based browsers (UC Browser and QQ Browser) due to a) lack of information, and b) the possible answer of #3875.

~~Depends on #3874, #3876, #3882, and #3884 to be merged first.~~ All have been merged.

## Data Sources
General:
https://en.wikipedia.org/wiki/Comparison_of_browser_engines

Chrome:
https://en.wikipedia.org/wiki/Google_Chrome_version_history
https://en.wikipedia.org/wiki/Blink_(browser_engine)

Firefox:
https://developer.mozilla.org/en-US/docs/Mozilla/Gecko/Versions

Safari:
https://en.wikipedia.org/wiki/Safari_version_history
https://www.somegeekintn.com/blog/stuff/iosvers/
https://developers.whatismybrowser.com/useragents/explore/software_name/safari/

IE/Edge:
https://en.wikipedia.org/wiki/Internet_Explorer
https://en.wikipedia.org/wiki/Trident_(software)
https://en.wikipedia.org/wiki/Microsoft_Edge
https://en.wikipedia.org/wiki/EdgeHTML

Opera:
https://en.wikipedia.org/wiki/Presto_(browser_engine)
https://en.wikipedia.org/wiki/Blink_(browser_engine)
https://github.com/mdn/browser-compat-data/pull/1712 (Android)
https://androidapksfree.com/devapk/opera/ (Android)

Node.js:
https://nodejs.org/en/download/releases/

Samsung Internet:
https://gist.github.com/poshaughnessy/5718717a04db20a02e9fdb3fc16e2258
https://medium.com/samsung-internet-dev/hello-samsung-internet-8-2-beta-521e4b215fb3
https://medium.com/samsung-internet-dev/8-2-stable-release-is-out-the-door-122cba80b788

Webview Android:
https://developers.whatismybrowser.com/useragents/explore/software_name/android-browser
https://developer.chrome.com/multidevice/webview/overview

### CODE REVIEW

File: api/Document.json [hunk 1]
```diff
@@ -8757,7 +8757,7 @@
               "version_added": false
             },
             "webview_android": {
-              "version_added": "37"
+              "version_added": "36"
```
Comment (jpmedley) [inline api/Document.json]: 36 is not possible. I'm not going to mark every instance, but please change this throughout.

Comment (vinyldarkscratch) [reply api/Document.json]: These are the changes in #3884.  I did find something that references the existence of a WebView v36 in an Android beta channel, and historically this repository has kept browsers that remained in beta.  Alternatively, I’m happy to remove it if it’s more trouble than it’s worth!

Comment (jpmedley) [reply api/Document.json]: It refers to a developer prelease. That's not something you would find in the wild, so to speak.

Comment (ExE-Boss) [reply api/Document.json]: Similarly, Firefox Android 11 - 13 only had beta releases, so their compat data isn’t stored in this repository either.

Comment (vinyldarkscratch) [reply api/Document.json]: Alright.  Scrapping those changes!

File: schemas/browsers.schema.json [hunk 2]
```diff
@@ -60,6 +60,15 @@
           "format": "uri",
           "description": "A link to the release notes or changelog for a given release."
         },
+        "engine": {
+          "type": "string",
+          "enum": ["Blink", "EdgeHTML", "Gecko", "Opera", "Presto", "Trident", "WebKit", "WebView", "V8"],
```
[the remark this answers is not in the document]
Comment (vinyldarkscratch) [reply schemas/browsers.schema.json]: I'm going to look into this one a little more and see if I can find more information about WebView.

**Comment by jpmedley** [comment]
For `webview_android` keep the engine. For webviews based on WebKit, list the webkit engine number not the corresponding Chrome version number. This is the information that best conveys compatibility.

**Comment by vinyldarkscratch** [comment]
I'm actually not sure about the pre-Chrome WebView versions, and what browser engines they were.  I'm planning on doing some research with some old emulators and seeing if I can get said data out.  Wish me luck!

**Comment by vinyldarkscratch** [comment]
In my search for Safari iOS' user agent strings, I found a database of practically every user agent string.  Within it, I was able to find WebView 2.0 through 4.2, and determine the WebKit versions used.  I was not able to find Android 1.x, so I shall leave that for whoever wants to claim it.

Overall, I'd say that once the final dependent PR is merged, this one's all ready to go!

**Comment by vinyldarkscratch** [comment]
This PR doesn’t set a release date for Chrome 75 beta, or really any browser, only adds an obvious engine number (Blink 75).  The purpose of this PR is to add engines to the browsers to help with feature parity.

**Comment by Elchi3** [comment]
Oh, and I think this PR shouldn't yet close https://github.com/mdn/browser-compat-data/issues/3705.

**Comment by vinyldarkscratch** [comment]
I agree with you on the Safari and WebKit versions, especially with the fact that Safari iOS 11.3 also had the same WebKit version as 12.  It's weird, and doesn't follow semantic versioning.  But what's weirder is that I have been able to confirm this was actually the case:

https://developers.whatismybrowser.com/useragents/parse/697853-safari-ios-iphone-webkit
https://developers.whatismybrowser.com/useragents/parse/746552-safari-ios-iphone-webkit
https://developers.whatismybrowser.com/useragents/parse/864604-safari-ios-ipad-webkit

Regarding Trident and IE, what I did with Opera before Presto was a thing is just add an engine, simply called "Opera", which matches the same version numbers as the browser.  We can probably do that for IE as well, or like you said, leave it blank.

Lastly, I'm no longer closing that issue with this PR!

**Comment by Elchi3** [comment]
Thanks for the links that confirm the Safari/WebKit versions. I can't test this myself due to no Apple devices at hand, but I think we can go with this for now as it seems correct (but weird).

Also thanks for the IE/Trident updates, this was an interesting history lesson. Leaving engines out for prior to IE8 seems like a good approach to me.

> Lastly, I'm no longer closing that issue with this PR!

Awesome, thanks!

**Pull request merged**